### PR TITLE
[PAY-1242] Mobile chats copy message button

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
Copy message button appears underneath message when opening reactions popup.

### Dragons

reactions overlap copy message button, waiting to hear back from desing.
Screenshot 2023-05-22 at 3 04 15 PM

### How Has This Been Tested?

local ios stage
Simulator Screen Shot - iPhone 14 Pro - 2023-05-22 at 15 13 07

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/screens/chat-screen/ChatScreen.tsx [hunk 1]
```diff
@@ -193,6 +193,8 @@ export const ChatScreen = () => {
   const composeRef = useRef<View | null>(null)
   const chatContainerRef = useRef<View | null>(null)
   const messageTop = useRef(0)
+  const messageWidth = useRef(0)
+  const messageHeight = useRef(0)
```
Comment (piazzatron) [inline packages/mobile/src/screens/chat-screen/ChatScreen.tsx:197]: Could combine into a single `messageDimensions` ref with `width` and `height`.

File: packages/mobile/src/screens/chat-screen/ReactionPopup.tsx [hunk 2]
```diff
@@ -134,7 +182,13 @@ export const ReactionPopup = ({
           { opacity: backgroundOpacityAnim.current }
         ]}
       />
-      <Pressable style={styles.outerPressable} onPress={handleClosePopup} />
+      <Pressable
+        style={styles.outerPressable}
+        onPress={() => {
+          console.log('parent press')
```
Comment (piazzatron) [inline packages/mobile/src/screens/chat-screen/ReactionPopup.tsx:188]: log

File: packages/mobile/src/screens/chat-screen/ReactionPopup.tsx [hunk 3]
```diff
@@ -149,23 +203,55 @@ export const ReactionPopup = ({
...
+                {
+                  top: messageTop - containerTop,
+                  right: isAuthor ? spacing(6) : undefined,
+                  left: !isAuthor ? spacing(6) : undefined
+                }
+              ]}
+            />
+          </View>
+        </Animated.View>
+        <Pressable
```
Comment (piazzatron) [inline packages/mobile/src/screens/chat-screen/ReactionPopup.tsx:232]: Would be worth it to pull out a `CopyButton` component here IMO, this render fn is getting pretty lengthy

File: packages/mobile/src/screens/chat-screen/ReactionPopup.tsx [hunk 4]
```diff
@@ -149,23 +203,55 @@ export const ReactionPopup = ({
...
-            isPopup={true}
+        <Pressable
+          style={[styles.innerPressable]}
+          onPress={() => {
+            console.log('inner press')
+            handleClosePopup()
+          }}
+        />
+        <Animated.View
+          style={{ opacity: otherOpacityAnim.current, zIndex: 40 }}
```
Comment (piazzatron) [inline packages/mobile/src/screens/chat-screen/ReactionPopup.tsx:214]: A comment explaining the Z index here would be 👌

File: packages/mobile/src/screens/chat-screen/ReactionPopup.tsx [hunk 5]
```diff
@@ -149,23 +203,55 @@ export const ReactionPopup = ({
...
+        >
+          <View>
```
Comment (piazzatron) [inline packages/mobile/src/screens/chat-screen/ReactionPopup.tsx:216]: Why'd we end up with this extra view?

**Review by piazzatron — APPROVED** [review]

File: packages/mobile/src/screens/chat-screen/ChatScreen.tsx (hunk 1 above)
Comment (dharit-tan) [reply packages/mobile/src/screens/chat-screen/ChatScreen.tsx:197]: actually no longer need width

File: packages/mobile/src/screens/chat-screen/ReactionPopup.tsx (hunk 5 above)
Comment (dharit-tan) [reply packages/mobile/src/screens/chat-screen/ReactionPopup.tsx]: Also not needed anymore!

**Pull request merged**